flink-streaming-java/src/main/java/org/apache/flink/streaming/util/retryable/AsyncRetryStrategies.java

```diff
@@ -0,0 +1,260 @@
+            this.resultPredicate = resultPredicate;
+            this.exceptionPredicate = exceptionPredicate;
+        }
+
+        @Override
+        public boolean canRetry(int currentAttempts) {
+            return currentAttempts <= maxAttempts;
+        }
+
+        @Override
+        public AsyncRetryPredicate<OUT> getRetryPredicate() {
+            return new AsyncRetryPredicate<OUT>() {
+                @Override
+                public Optional<Predicate<Collection<OUT>>> resultPredicate() {
+                    if (null == resultPredicate) {
+                        return Optional.empty();
+                    }
+                    return Optional.of(resultPredicate);
+                }
+
+                @Override
+                public Optional<Predicate<Throwable>> exceptionPredicate() {
+                    if (null == exceptionPredicate) {
+                        return Optional.empty();
+                    }
+                    return Optional.of(exceptionPredicate);
+                }
+            };
+        }
+
+        @Override
+        public long getBackoffTimeMillis(int currentAttempts) {
+            return backoffTimeMillis;
+        }
+    }
+
+    /** FixedDelayRetryStrategyBuilder for building a FixedDelayRetryStrategy. */
+    public static class FixedDelayRetryStrategyBuilder<OUT> {
+        private int maxAttempts;
+        private long backoffTimeMillis;
+        private Predicate<Collection<OUT>> resultPredicate;
+        private Predicate<Throwable> exceptionPredicate;
+
+        public FixedDelayRetryStrategyBuilder(int maxAttempts, long backoffTimeMillis) {
+            Preconditions.checkArgument(
+                    maxAttempts > 0, "maxAttempts should be greater than zero.");
+            Preconditions.checkArgument(
+                    backoffTimeMillis > 0, "backoffTimeMillis should be greater than zero.");
+            this.maxAttempts = maxAttempts;
+            this.backoffTimeMillis = backoffTimeMillis;
+        }
+
+        public FixedDelayRetryStrategyBuilder ifResult(
+                @Nonnull Predicate<Collection<OUT>> resultRetryPredicate) {
+            this.resultPredicate = resultRetryPredicate;
+            return this;
+        }
+
+        public FixedDelayRetryStrategyBuilder ifException(
+                @Nonnull Predicate<Throwable> exceptionRetryPredicate) {
+            this.exceptionPredicate = exceptionRetryPredicate;
+            return this;
+        }
+
+        public FixedDelayRetryStrategy build() {
+            return new FixedDelayRetryStrategy(
+                    maxAttempts, backoffTimeMillis, resultPredicate, exceptionPredicate);
+        }
+    }
+
+    /** ExponentialBackoffDelayRetryStrategy. */
+    public static class ExponentialBackoffDelayRetryStrategy<OUT>
+            implements AsyncRetryStrategy<OUT> {
+        private static final long serialVersionUID = 1L;
+        private final int maxAttempts;
+
+        private final long initialDelay;
```

Comment:
`initialDelay` is not necessary as a class variable?

Reply:
You're right, `initialDelay` can be omitted here.